content/docs/concepts/components/lifecycle/detector.md

```diff
@@ -0,0 +1,56 @@
+One of the [input files][inputs] is [`order.toml`][order] and two of its [output files][outputs] are [`group.toml`][group] and [`plan.toml`][plan].
+
+Unless some flags are passed in, the detector will use the following defaults:
+* OrderPath: `/cnb/order.toml`
+* GroupPath: `<layers>/group.toml`
+* PlanPath: `<layers>/plan.toml`
+The full list of flags and their defaults can be found [here][detector].
+
+#### `order.toml`
+`order.toml` contains a list of groups. Each group contains a list of buildpacks.
+The detector reads `order.toml` and looks for the first group that passes the detection process.
+If all of the groups fail, the detection process fails.
+Each buildpack in each group is marked as either optional or as required.
+In order to pass the detection process, two conditions should be satisfied:
+The detect scripts of all required buildpacks should pass successfully (the exit code is zero).
+The detector should be able to create a build plan with all of the requirements of the group’s buildpacks.
+The first group that passes both steps is written to `group.toml` and its build plan is written to `plan.toml`.
+Note: if the detect script of an optional buildpack failed, the group can still pass the detection process and be the “chosen”  group. In order for that to happen, there should be at least one (required or optional) buildpack in this group that passes the detect script successfully.
+
+#### `group.toml`
+The buildpacks of the “chosen” group will be written to `group.toml` if they pass the step of creating `plan.toml`. They are written in the same order as they appear in `order.toml` after filtering out all of the (optional) buildpacks whose detect script failed.
+
+#### `plan.toml`
+Each buildpack can define two lists with provided and required dependencies (or several pairs of lists separated by `or`). These lists (if they aren’t empty) are called a [build plan][buildPlan] and they are part of the [output of each buildpack’s detect script][detectScriptOutput].
+The detector reads the build plans of the buildpacks of the “chosen” group (after filtering out the buildpacks whose detect script failed). It goes over all of the options and tries to create a file with a list of entries, each with provides and requires lists, that fulfills all of the buildpacks requirements. This file is called `plan.toml`.
+What do these lists need to accomplish:
+A dependency that is provided by a buildpack, must be required by either the buildpack itself or a later buildpack in the group.
+A dependency that is required by a buildpack, must be provided by the buildpack itself or a previous buildpack in the group.
+If at least one of the above failed on a required buildpack, the trial fails and the detector will look for the next trial. If at least one of the above failed on an optional buildpack, this buildpack should be excluded from the final plan. If all of the trials fail, the group fails (and the detector moves on to the next group.
```

Comment:
Add an explanation on the order of the trials when using meta buildpacks (ordered buildpacks).
Give an example of what it means to run the trials "left-to-right depth-first".